ui/src/app/views/workspace-list/component.ts

```diff
@@ -74,6 +79,18 @@ export class WorkspaceListComponent implements OnInit {
             });
   }
 
+  queryBillingStatus(): void {
+    this.profileService.getMe().subscribe((profile: Profile) => {
+      if (profile.freeTierBillingProjectStatus === BillingProjectStatus.Ready) {
+        this.billingProjectInitialized = true;
+      } else {
+        setTimeout(() => {
+          this.queryBillingStatus();
+        }, 10000);
```

Comment:
Is this timeout never cleared? In other words, can I launch a DOS against FireCloud simply by visiting this page enough times in a row?